File: pdu/python/geometry_msgs/pdu_cdr_conv_Transform.py
from .pdu_pytype_Transform import Transform
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..geometry_msgs.pdu_cdr_conv_Quaternion import *
from ..geometry_msgs.pdu_cdr_conv_Vector3 import *



def py_to_cdr_body_Transform(writer: CdrWriter, src: Transform):
    py_to_cdr_body_Vector3(writer, src.translation)
    py_to_cdr_body_Quaternion(writer, src.rotation)


def cdr_body_to_py_Transform(reader: CdrReader, dst: Transform):
    cdr_body_to_py_Vector3(reader, dst.translation)
    cdr_body_to_py_Quaternion(reader, dst.rotation)
    return dst


def py_to_cdr_Transform(src: Transform) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_Transform(writer, src)
    return writer.bytes()


def cdr_to_py_Transform(cdr_payload) -> Transform:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = Transform()
    return cdr_body_to_py_Transform(reader, dst)

File: pdu/javascript/geometry_msgs/pdu_cdr_conv_Transform.js
import { Transform } from './pdu_jstype_Transform.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Quaternion } from '../geometry_msgs/pdu_jstype_Quaternion.js';
import { PduQuaternionConverter } from '../geometry_msgs/pdu_cdr_conv_Quaternion.js';
import { Vector3 } from '../geometry_msgs/pdu_jstype_Vector3.js';
import { PduVector3Converter } from '../geometry_msgs/pdu_cdr_conv_Vector3.js';


export class PduTransformConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { Transform } src
     */
    static to_cdr_body(writer, src) {
        PduVector3Converter.to_cdr_body(writer, src.translation);
        PduQuaternionConverter.to_cdr_body(writer, src.rotation);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { Transform } dst
     * @returns { Transform }
     */
    static cdr_body_to_js(reader, dst) {
        PduVector3Converter.cdr_body_to_js(reader, dst.translation);
        PduQuaternionConverter.cdr_body_to_js(reader, dst.rotation);
        return dst;
    }

    /**
     * @param { Transform } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { Transform }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new Transform());
    }
}
